echoregions/regions2d/regions2d.py

```diff
@@ -73,42 +73,140 @@ def to_json(self, save_path: str = None) -> None:
+            if isinstance(time_range, List):
+                if len(time_range) == 2:
+                    if isinstance(time_range[0], Timestamp) and isinstance(
+                        time_range[1], Timestamp
+                    ):
+                        if time_range[0] < time_range[1]:
+                            if region is None:
+                                region = self.data
+                            for index, row in region.iterrows():
+                                remove_row = False
+                                for time in row["time"]:
+                                    if time_range[0] > Timestamp(time) or time_range[
+                                        1
+                                    ] < Timestamp(time):
+                                        remove_row = True
+                                if remove_row:
+                                    region.drop(index, inplace=True)
+                            untouched = False
+                        else:
+                            raise ValueError(
+                                f"1st index value must be later than 0th index \
+                                             value. Currently 0th index value is {time_range[0]} \
+                                             and 1st index value is {time_range[1]}"
+                            )
+                    else:
+                        raise TypeError(
+                            f"Invalid time_range value types: \
+                                        {type(time_range[0])} and {type(time_range[1])}. Must \
+                                        be both of type Timestamp."
+                        )
+                else:
+                    raise ValueError(
+                        f"Invalid time_range size: {len(time_range)}. \
+                        Must be of size 2."
+                    )
+            else:
+                raise TypeError(
+                    f"Invalid time_range type: {type(time_range)}. Must be \
+                                of type List."
+                )
+        if depth_range is not None:
+            if isinstance(depth_range, List):
+                if len(depth_range) == 2:
+                    if isinstance(depth_range[0], (float, int)) and isinstance(
+                        depth_range[1], (float, int)
+                    ):
+                        if depth_range[0] < depth_range[1]:
+                            if region is None:
+                                region = self.data
+                            for index, row in region.iterrows():
+                                remove_row = False
+                                for depth in row["depth"]:
+                                    if depth_range[0] > depth or depth_range[1] < depth:
+                                        remove_row = True
+                                if remove_row:
+                                    region.drop(index, inplace=True)
```

Comment:
Similar comments as before for `time_range` re .apply` function and use `regions` assigned upfront.